File: test/fixtures/ruby/post_image.rb
require 'net/http'

uri = URI('http://localhost:28139/targetservice')
req = Net::HTTP::Post.new(uri)

req.set_form(
  [
    [
      'image',
      File.open('image.jpg')
    ]
  ],
  'multipart/form-data'
)

req_options = {
  use_ssl: uri.scheme == 'https'
}
res = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
  http.request(req)
end

File: test/fixtures/csharp/post_image.cs
using System.Net.Http;

HttpClient client = new HttpClient();

HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, "http://localhost:28139/targetservice");

MultipartFormDataContent content = new MultipartFormDataContent();
content.Add(new ByteArrayContent(File.ReadAllBytes("image.jpg")), "image", Path.GetFileName("image.jpg"));
request.Content = content;

HttpResponseMessage response = await client.SendAsync(request);
response.EnsureSuccessStatusCode();
string responseBody = await response.Content.ReadAsStringAsync();
